packages/google-pay-integration/src/gateways/google-pay-tdonlinemart-gateway.ts

```diff
@@ -1,9 +1,51 @@
-import { PaymentIntegrationService } from '@bigcommerce/checkout-sdk/payment-integration-api';
+import { FormPoster } from '@bigcommerce/form-poster';
+
+import {
+    getBrowserInfo,
+    PaymentArgumentInvalidError,
+    PaymentIntegrationService,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import { isTdOnlineMartAdditionalAction } from '../guards/is-google-pay-td-online-mart-additional-action';
+import { ExtraPaymentData, TdOnlineMartThreeDSErrorBody } from '../types';
 
 import GooglePayGateway from './google-pay-gateway';
 
 export default class GooglePayTdOnlineMartGateway extends GooglePayGateway {
-    constructor(service: PaymentIntegrationService) {
+    constructor(service: PaymentIntegrationService, private formPoster: FormPoster) {
         super('worldlinena', service);
     }
+
+    async extraPaymentData(): Promise<undefined | ExtraPaymentData> {
```

Comment:
Why do you set `undefined` as the returning type? There is no need to return `undefined` as one of possible returning types